lib/healthcheck/index.js

```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+var fs        = require("fs");
+var path      = require("path");
+var basename  = path.basename(module.filename);
+var logger    = require("../logger.js").getLogger({"module": __filename});
+
+module.exports = function () {
+  var reportStart = process.hrtime();
+  var report = {
+    report_as_of: new Date().toISOString(),
+    duration_millis: 0,
+    tests: []
+  };
+
+  // Load up an array of promises for each (other) file in this directory
+  // All tests should resolve and never reject to avoid breaking other tests
+  var tests = [];
+  fs.readdirSync(__dirname)
+    .filter(function(file) {
+      return (file.indexOf(".") !== 0) && (file !== basename);
+    })
+    .forEach(function(file) {
+      logger.info("Running test: " + file);
+      tests.push(require("./" + file)());
+    });
+
+  // Once all promises have resolved, return the report
+  return Promise.all(tests)
+      .then(function (results) {
+        report.tests = results;
+        finalizeReport(report, reportStart);
+        return report;
+      }).catch(function (results) {
+        report.tests = results;
+        finalizeReport(report, reportStart);
+        return report;
+      });
+};
+
+function finalizeReport(report, reportStart) {
+  report.duration_millis = elapsed_time(reportStart);
```

Comment:
I'd be cool with inlining this since there are now 2 invocations and not 6. Not important.